### PR TITLE
Move term package to component base and switch to moby/term

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/go.mod [hunk 1]
```diff
@@ -5,38 +5,35 @@ module k8s.io/kubectl
 go 1.13
 
 require (
-	github.com/Azure/go-ansiterm v0.0.0-20170929234023-d6e3b3328b78 // indirect
 	github.com/MakeNowJust/heredoc v0.0.0-20170808103936-bb23615498cd
 	github.com/chai2010/gettext-go v0.0.0-20160711120539-c6fed771bfd5
 	github.com/davecgh/go-spew v1.1.1
 	github.com/daviddengcn/go-colortext v0.0.0-20160507010035-511bcaf42ccd
 	github.com/docker/distribution v2.7.1+incompatible
-	github.com/docker/docker v0.7.3-0.20190327010347-be7ac8be2ae0
```
Comment (liggitt) [inline staging/src/k8s.io/kubectl/go.mod]: 💯 for dropping this from kubectl deps

**Comment by dims** [comment]
@liggitt can you please re-apply lgtm? (had to rebase to master)

**Comment by luxas** [comment]
Thanks a lot @dims!
Is there any possibility to get this into the v1.18.x branch? I can't vendor `k8s.io/kubectl` at all because of this:

```
go: github.com/<redacted>/pkg/kubernetes/drain imports
        k8s.io/kubectl/pkg/drain imports
        k8s.io/kubectl/pkg/cmd/util imports
        k8s.io/kubectl/pkg/util/templates imports
        k8s.io/kubectl/pkg/util/term imports
        github.com/docker/docker/pkg/term imports
        github.com/docker/docker/pkg/term/windows imports
        github.com/Sirupsen/logrus: github.com/Sirupsen/logrus@v1.6.0: parsing go.mod:
        module declares its path as: github.com/sirupsen/logrus
                but was required as: github.com/Sirupsen/logrus
```

Forcing a kubectl import of version `v1.19.0-alpha.1` (with this patch) works though.

**Comment by liggitt** [comment]
@luxas make sure you are using the version of github.com/docker/docker k8s.io/kubectl@v0.18.x expects (be7ac8be2ae0)... this should work properly

**Comment by luxas** [comment]
:facepalm: thanks!
I _thought_ I did this when I tried different configurations, but as you could tell I somehow didn't :sweat_smile: 
Cheers!

**Pull request merged**